v1/logging/reporter.go

```diff
@@ -0,0 +1,109 @@
+// logging implements a logs reporter.
+package logging
+
+import (
+	"context"
+
+	"gopkg.in/inconshreveable/log15.v2"
+)
+
+// Reporter represents a logging reporter instance.
+type Reporter struct {
+	log log15.Logger
+
+	config *Config
+}
+
+// New returns a new logging reporter instance.
+func New(config *Config) (*Reporter, error) {
+	var reporter Reporter
+
+	if config == nil {
+		return nil, nil
+	}
+
+	if err := config.validate(); err != nil {
+		return nil, err
+	}
+	reporter.config = config
+
+	reporter.log = log15.New()
+	reporter.log.SetHandler(log15.DiscardHandler())
+
+	handlers := make([]log15.Handler, 0)
```

Comment:
The cap can be `len(reporter.config.Destinations)`. But not mandatory in that case.

Reply:
What good does it do?

Reply:
When you will call `append` size is already allocated in memory. So it will not alloc at each tour of for.
But if it's not a hot part of the program it's not mandatory.
Hope to you to determine if it's needed :)

Reply:
Thank you for the suggestion, but I'll pass on this one as it adds extra complexity to a case where the performance requirement doesn't warrant it 😉